src/app/models/vehicle-tech-record.model.ts

```diff
@@ -1,11 +1,134 @@
 export interface VehicleTechRecordModel {
-  vrms: VrmModel[];
+  primaryVrm: string;
+  secondaryVrms: string[];
   vin: string;
   trailerId?: string;
   systemNumber: string;
+  techRecord: TechRecordModel[]
 }
 
-export interface VrmModel {
-  vrm: string;
-  isPrimary: boolean;
+export enum StatusCodes {
+  ARCHIVED = 'archived',
+  CURRENT = 'current',
+  PROVISIONAL = 'provisional'
+}
+
+export enum VehicleTypes {
+  PSV = 'psv',
+  HGV = 'hgv',
+  TRL = 'trl'
+}
+
+export enum FuelTypes {
+  DIESELPETROL = 'DieselPetrol',
+  HYBRID = 'Hybrid',
+  ELECTRIC = 'Electric',
+  CNG = 'CNG',
+  FUELCELL = 'Fuel cell',
+  LNG = 'LNG',
+  OTHER = 'Other'
+}
+
+export enum VehicleClassDescriptions {
+  MOTORBIKE_OVER_200CC = 'motorbikes over 200cc or with a sidecar',
+  NOT_APPLICABLE = 'not applicable',
+  SMALL_PSV = 'small psv (ie: less than or equal to 22 seats)',
+  MOTORBIKE_UPTO_200CC = 'motorbikes up to 200cc',
+  TRAILER = 'trailer',
+  LARGE_PSV = 'large psv(ie: greater than 23 seats)',
+  THREE_WHEELER = '3 wheelers',
+  HGV = 'heavy goods vehicle',
+  MOT_CLASS_4 = 'MOT class 4',
+  MOT_CLASS_5 = 'MOT class 5',
+  MOT_CLASS_7 = 'MOT class 7'
+}
+
+export enum VehicleConfigurations {
+  RIGID = 'rigid',
+  ARTICULATED = 'articulated',
+  CENTRE_AXLE_DRAWBAR = 'centre axle drawbar',
+  SEMI_CAR_TRANSPORTER = 'semi-car transporter',
+  SEMI_TRAILER = 'semi-trailer',
+  LOW_LOADER = 'low loader',
+  OTHER = 'other',
+  DRAWBAR = 'drawbar',
+  FOUR_IN_LINE = 'four-in-line',
+  DOLLY = 'dolly',
+  FULL_DRAWBAR = 'full drawbar'
+}
+
+export enum FrameDescriptions {
+  CHANNEL_SECTION = 'Channel section',
+  SPACE_FRAME = 'Space frame',
+  I_SECTION = 'I section',
+  TUBULAR = 'Tubular',
+  FRAME_SECTION = 'Frame section',
+  OTHER = 'Other',
+  INTEGRAL = 'integral',
+  BOX_SECTION = 'Box section',
+  U_SECTION = 'U section'
+}
+
+export enum EuVehicleCategories {
+  M1 = 'm1',
+  M2 = 'm2',
+  M3 = 'm3',
+  n1 = 'n1',
+  n2 = 'n2',
+  n3 = 'n3',
+  o1 = 'o1',
+  o2 = 'o2',
+  o3 = 'o3',
+  o4 = 'o4',
+  l1e_a = 'l1e-a',
+  l1e = 'l1e',
+  l2e = 'l2e',
+  l3e = 'l3e',
+  l4e = 'l4e',
+  l5e = 'l5e',
+  l6e = 'l6e',
+  l7e = 'l7e'
+}
+
+export enum VehicleSizes {
+  SMALL = 'small',
+  LARGE = 'large'
+}
+
+export interface TechRecordModel {
```

Comment:
Do we want a smarter model than this? Some of these fields are not applicable for certain vehicle types, do we want to model each vehicle type as its own class? They can all share a common base class

Reply:
I think we're kind of using this as a generic 'what we can expect back from the DynamoDB'. But I guess in the get request to the DB you could specify the return type for the response to be something like `<PsvTechRecordModel | HgvTechRecordModel | TrlTechRecordModel | ...>` but maybe that's a bit too much? Especially when we add additional vehicle types.

Our vehicle specific type 'classes' seem to be based in the templates we have for the dynamic forms where we have separate ones for PSV, HGV and TRL.

Reply:
I think `<PsvTechRecordModel | HgvTechRecordModel | TrlTechRecordModel | ...>` will be exactly what we need, especially if we want type safety and logical consistency. What comes from DynamoDb doesn't have to define our classes in Angular (we can always map the data coming back to new types). But maybe this is something we can build later.